File: src/java/main/java/au/edu/uq/cai/TissueStack/dataobjects/Configuration.java
/*
 * This file is part of TissueStack.
 *
 * TissueStack is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * TissueStack is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with TissueStack.  If not, see <http://www.gnu.org/licenses/>.
 */
package au.edu.uq.cai.TissueStack.dataobjects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@Entity
@Table(name="configuration")
@XmlRootElement(name="Configuration", namespace=IGlobalConstants.XML_NAMESPACE)
public class Configuration {
	private String name;
	private String value;
	private String description;
	
	@Id
	@Column(name="name")
	@XmlElement(name="Name", namespace=IGlobalConstants.XML_NAMESPACE)
	public String getName() {
		return name;
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	@Column(name="value")
	@XmlElement(name="Value", namespace=IGlobalConstants.XML_NAMESPACE)
	public String getValue() {
		return value;
	}
	
	public void setValue(String value) {
		this.value = value;
	}
	
	@Column(name="description")
	@XmlElement(name="Description", namespace=IGlobalConstants.XML_NAMESPACE)
	public String getDescription() {
		return description;
	}
	
	public void setDescription(String description) {
		this.description = description;
	}
}

File: src/c++/database/Configuration.cpp
/*
 * This file is part of TissueStack.
 *
 * TissueStack is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * TissueStack is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with TissueStack.  If not, see <http://www.gnu.org/licenses/>.
 */
#include "database.h"

tissuestack::database::Configuration::Configuration(
		const std::string name, const std::string value, const std::string description) :
		_name(name), _value(value), _description(description) {
	if (name.empty() || value.empty())
		THROW_TS_EXCEPTION(tissuestack::common::TissueStackApplicationException,
								"Configuration Data Object needs a non-empty name/value pair!");
}

const std::string tissuestack::database::Configuration::getName() const
{
	return this->_name;
}

const std::string tissuestack::database::Configuration::getValue() const
{
	return this->_value;
}

const std::string tissuestack::database::Configuration::getDescription() const
{
	return this->_description;
}

void tissuestack::database::Configuration::setValue(const std::string value)
{
	if (value.empty()) return;

	this->_value = value;
}


const std::string tissuestack::database::Configuration::getJson() const
{
	std::ostringstream json;
	json << "{ \"name\": \"" << tissuestack::utils::Misc::maskQuotesInJson(this->_name) << "\"";
	json << ", \"value\": \"" << tissuestack::utils::Misc::maskQuotesInJson(this->_value) << "\"";
	if (!this->_description.empty())
		json << ", \"description\": \"" << tissuestack::utils::Misc::maskQuotesInJson(this->_description) << "\"";
	json << " }";

	return json.str();
}
